ReSharper.FSharp/src/FSharp.Common/src/Util/FSharpPredefinedType.fs

```diff
@@ -149,3 +149,11 @@ let isUnit (fcsType: FSharpType) =
 let isNativePtr (fcsType: FSharpType) =
     try fcsType.StrippedType.BasicQualifiedName = StandardTypeNames.IntPtr
     with _ -> false
+
+let hasPureGetter (fcsType: FSharpType) =
+    fcsType.HasTypeDefinition && fcsType.TypeDefinition.IsArrayType || 
+
+    let name = fcsType.StrippedType.BasicQualifiedName
+    name.StartsWith("Microsoft.FSharp.Collections.FSharpList`", StringComparison.Ordinal) ||
+    name.StartsWith("System.Collections.Generic.List`", StringComparison.Ordinal) ||
+    name.StartsWith("System.String", StringComparison.Ordinal)
```

Comment:
You can also use `CommonUtil.startsWith` here. Though, I'm becoming a bit less enthusiastic about such helpers over time, as it's impossible to discover them and it only adds a second way to do things as a result. On the other hand, these helpers sometimes make it much more readable.